cs.c

```diff
@@ -584,9 +591,14 @@ size_t CAPSTONE_API cs_disasm(csh ud, const uint8_t *buffer, size_t size, uint64
 			insn_cache->address = offset;
 			insn_cache->size = (uint16_t)skipdata_bytes;
 			memcpy(insn_cache->bytes, buffer, skipdata_bytes);
+#ifdef CAPSTONE_DIET
+ 			insn_cache->mnemonic[0] = '\0';
```

Comment:
From what I can tell you do not, and the memory is uninitialised here otherwise. (insn_cache → total → cs_mem_malloc).